fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -233,69 +237,60 @@ protected RdfStream getResourceTriples(final int limit) {
 
         final LdpPreferTag ldpPreferences = new LdpPreferTag(returnPreference);
 
-        final RdfStream rdfStream = new RdfStream();
-
         final Predicate<Triple> tripleFilter = ldpPreferences.prefersServerManaged() ? x -> true :
             IS_MANAGED_TRIPLE.negate();
 
-        if (ldpPreferences.prefersServerManaged()) {
-            rdfStream.concat(getTriples(LdpRdfContext.class));
-        }
-
-        rdfStream.concat(filter(getTriples(TypeRdfContext.class), tripleFilter::test));
+        final List<Stream<Triple>> streams = new ArrayList<>();
 
-        rdfStream.concat(filter(getTriples(PropertiesRdfContext.class), tripleFilter::test));
+        streams.add(getTriples(PROPERTIES).filter(tripleFilter));
 
-        if (!returnPreference.getValue().equals("minimal")) {
+        if (returnPreference.getValue().equals("minimal")) {
+            if (ldpPreferences.prefersServerManaged()) {
+                streams.add(getTriples(SERVER_MANAGED)
```

Comment:
I just want to add that, while I would rather not release before moving
from enum to interface, I am personally swayed by the PR growing too large
to rebase if we need to merge and spin new tickets off.

On Wed, Feb 17, 2016 at 2:43 PM, Aaron Coburn <notifications@github.com>
wrote:

> In
> fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java
> <https://github.com/fcrepo4/fcrepo4/pull/962#discussion_r53219510>:
>
> >
> > -        if (!returnPreference.getValue().equals("minimal")) {
> > +        if (returnPreference.getValue().equals("minimal")) {
> > +            if (ldpPreferences.prefersServerManaged()) {
> > +                streams.add(getTriples(SERVER_MANAGED)
>
> The simple way to handle that is to add a MINIMAL enum value that would
> allow for those permutations. It's not the most elegant design, but it
> would work and could be replaced with a stronger `interface`-based design
> later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fcrepo4/fcrepo4/pull/962/files#r53219510>.
>